client/components/domains/register-domain-step/index.jsx

```diff
@@ -584,20 +583,36 @@ const RegisterDomainStep = React.createClass( {
 				}
 				break;
 
-			case 'mappable_but_forbidden_subdomain':
+			case 'not_mappable_forbidden_subdomain':
 				message = this.translate( 'Subdomains starting with \'www.\' cannot be mapped to a WordPress.com blog' );
 				break;
 
-			case 'mappable_but_mapped_domain':
+			case 'not_mappable_invalid_tld':
+			case 'invalid_query':
+				message = this.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
+					args: { domain: domain }
+				} );
+				break;
+
+			case 'not_mappable_mapped_domain':
 				message = this.translate( 'This domain is already mapped to a WordPress.com site.' );
 				break;
 
-			case 'mappable_but_restricted_domain':
+			case 'not_mappable_restricted_domain':
 				message = this.translate(
 					'You cannot map another WordPress.com subdomain - try creating a new site or one of the custom domains below.'
 				);
 				break;
 
+			case 'not_mappable_recently_mapped':
+				message = this.translate( 'This domain has been recently mapped by a different user - please try again later.' );
+				break;
+
+			// Generic error message when domain is not mappable without an explicit unmappability reason
+			case 'not_mappable':
+				message = this.translate( 'This domain cannot be mapped.' );
```

Comment:
This case should never be reached (it's currently impossible for the backend to return `! is_mappable` without an unmappability reason), but I've added it just in case - so it's sounding very generic. I can add the sorry prefix, but I usually reserve those for catastrophic errors when we are clearly at fault. But since we use that form rather freely in other parts of domains, I can comply :)